sdk/cosmos/azure_data_cosmos/src/partition_key.rs

```diff
@@ -90,8 +94,11 @@ impl<T: Into<PartitionKey>> From<T> for QueryPartitionStrategy {
 #[derive(Debug, Clone)]
 pub struct PartitionKey(Vec<PartitionKeyValue>);
 
-impl PartitionKey {
-    pub(crate) fn into_header_value(self) -> azure_core::Result<String> {
+impl AsHeaders for PartitionKey {
+    type Error = azure_core::Error;
+    type Iter = std::iter::Once<(HeaderName, HeaderValue)>;
+
+    fn as_headers(&self) -> Result<std::iter::Once<(HeaderName, HeaderValue)>, Self::Error> {
```

Comment:
Curious, could/should this be:

```suggestion
    fn as_headers(&self) -> Result<Self::Iter, Self::Error> {
```

Reply:
Yep, it should be. The two are identical, but it's generally preferred to keep the actual type in one place and just reference it (because duplication yields inconsistency ;)).

I _wanted_ to use an `impl Trait` type to avoid having to name the actual type, but support for `impl Trait` in associated types is unstable 😞